secret-engine/src/main/java/com/sequenceiq/cloudbreak/service/secret/service/VaultRetryService.java

```diff
@@ -48,12 +67,34 @@ public VaultRetryService(MetricService metricService) {
                     maxDelayExpression = "${vault.retry.maxdelay:10000}")
     )
     public <T> T tryWritingVault(Supplier<T> action) throws Retry.ActionFailedException {
-        try {
-            return action.get();
-        } catch (RuntimeException e) {
-            LOGGER.error("Exception during writing vault", e);
-            metricService.incrementMetricCounter(MetricType.VAULT_READ_FAILED);
-            throw new Retry.ActionFailedException(e.getMessage());
+
+        Optional<Span> optionalSpan = initSpan("write");
+        return optionalSpan.map(span -> {
+            try (Scope ignored = tracer.activateSpan(span)) {
+                try {
+                    return action.get();
+                } catch (RuntimeException e) {
+                    LOGGER.error("Exception during writing vault", e);
+                    span.setTag(TracingUtil.ERROR, true);
+                    span.setTag(TracingUtil.MESSAGE, e.getMessage());
+                    metricService.incrementMetricCounter(MetricType.VAULT_READ_FAILED);
+                    throw new Retry.ActionFailedException(e.getMessage());
+                }
+            } finally {
+                span.finish();
+            }
+        }).orElseGet(action::get);
+    }
+
+    private Optional<Span> initSpan(String operationType) {
+        Span activeSpan = tracer.activeSpan();
```

Comment:
this function could be more compact with Optional.ofNullable(...).map(...)

Reply:
ok